packages/plugin-core/src/commands/VaultAddCommand.ts

```diff
@@ -182,8 +259,85 @@ export class VaultAddCommand extends BasicCommand<CommandOpts, CommandOutput> {
     return { vaults, workspace };
   }
 
+  async handleRemoteRepoSelfContained(
+    opts: CommandOpts
+  ): Promise<{ vaults: DVault[] }> {
+    return window.withProgress(
+      {
+        location: ProgressLocation.Notification,
+        title: "Adding remote vault",
+        cancellable: false,
+      },
+      async (progress) => {
+        const { wsRoot } = ExtensionProvider.getDWorkspace();
+        progress.report({
+          message: "cloning repo",
+          increment: 0,
+        });
+        const { name, pathRemote: remoteUrl } = opts;
+        const localUrl = path.join(wsRoot, opts.path);
+        if (!remoteUrl) {
+          throw new DendronError({
+            message:
+              "Remote vault has no remote set. This should never happen, please send a bug report if you encounter this.",
+          });
+        }
+
+        await fs.ensureDir(localUrl);
+        const git = new Git({ localUrl, remoteUrl });
```

Comment:
is there a reason we don't use simplegit as in `handleRemoteRepo`? also, does it make sense to encapsulate common clone logic across handle remote functions?

Reply:
I think we should either deprecate our custom git wrapper and move towards simplegit, or deprecate simplegit and move towards our custom wrapper. I avoided simeplegit because I'm leaning more towards the custom wrapper, but that's mostly because I wrote a lot of code for it :sweat_smile: 

I don't think we need to encapsulate the common logic, because we're planning to deprecate and remove support for old style vaults (at the very least creating of old style vaults). So the duplicated code will eventually be removed.